modules/tinymce/src/themes/silver/main/ts/ui/view/ViewWrapper.ts

```diff
@@ -0,0 +1,180 @@
+import { ViewConfig } from './ViewTypes';
+
+const makeViews = (parts: SlotContainerTypes.SlotContainerParts, viewConfigs: ViewConfig, providers: UiFactoryBackstageProviders) => {
+  return Obj.mapToArray(viewConfigs, (config, name) => {
+    const internalViewConfig: BridgeView.View = StructureSchema.getOrDie(BridgeView.createView(config));
+
+    return parts.slot(name, View.sketch({
+      dom: {
+        tag: 'div',
+        classes: [ 'tox-view' ]
+      },
+      viewConfig: internalViewConfig,
+      components: [
+        ...internalViewConfig.buttons.length > 0 ? [
+          View.parts.header({
+            buttons: internalViewConfig.buttons,
+            providers
+          })
+        ] : [],
+        View.parts.pane({})
+      ]
+    }));
+  });
+};
+
+const makeSlotContainer = (viewConfigs: ViewConfig, providers: UiFactoryBackstageProviders) => SlotContainer.sketch((parts) => ({
+  dom: {
+    tag: 'div',
+    classes: [ 'tox-view-wrap__slot-container' ]
+  },
+  components: makeViews(parts, viewConfigs, providers),
+  slotBehaviours: SimpleBehaviours.unnamedEvents([
+    AlloyEvents.runOnAttached((slotContainer) => SlotContainer.hideAllSlots(slotContainer))
+  ])
+}));
+
+const getCurrentName = (slotContainer: AlloyComponent) => {
+  return Arr.find(SlotContainer.getSlotNames(slotContainer), (name) =>
+    SlotContainer.isShowing(slotContainer, name)
+  );
+};
+
+const hideContainer = (comp: AlloyComponent) => {
+  const element = comp.element;
+  Css.set(element, 'display', 'none');
+  Attribute.set(element, 'aria-hidden', 'true');
+};
+
+const showContainer = (comp: AlloyComponent) => {
+  const element = comp.element;
+  Css.remove(element, 'display');
+  Attribute.remove(element, 'aria-hidden');
+};
+
+const makeViewInstanceApi = (slot: HTMLElement): BridgeView.ViewInstanceApi => ({
+  getContainer: Fun.constant(slot)
+});
+
+const runOnShow = (slotContainer: AlloyComponent, name: string) => {
+  SlotContainer.getSlot(slotContainer, name).each((view) => {
+    View.getPane(view).each((pane) => {
+      const onShow = View.getOnShow(view);
+      onShow(makeViewInstanceApi(pane.element.dom));
+    });
+  });
+};
+
+const runOnHide = (slotContainer: AlloyComponent, name: string) => {
+  SlotContainer.getSlot(slotContainer, name).each((view) => {
+    View.getPane(view).each((pane) => {
+      const onHide = View.getOnHide(view);
+      onHide(makeViewInstanceApi(pane.element.dom));
+    });
+  });
+};
+
+interface SilverViewWrapperSpec extends Sketcher.SingleSketchSpec {
+  backstage: UiFactoryBackstage;
+}
+
+interface SilverViewWrapperDetail extends Sketcher.SingleSketchDetail {
+  uid: string;
+  backstage: UiFactoryBackstage;
+}
+
+interface SilverViewWrapperApis {
+  setViews: (comp: AlloyComponent, viewConfigs: ViewConfig) => void;
+  whichView: (comp: AlloyComponent) => Optional<string>;
+  toggleView: (comp: AlloyComponent, showMainView: () => void, hideMainView: () => void, name: string) => boolean;
+}
```

Comment:
Readonly? Also these should be at the top as we've always preferred have types at the top not in the middle.

- [ ] Readonly types and move